mqtt/docs/persistance.md

```diff
@@ -0,0 +1,24 @@
+# Broker Persistance
+The broker's persistance strategy is modeled after mosquitto's. It saves the entire state to one compressed file. File saving is triggered by a configurable timer and on shutdown.
+
+## Requirements
+* Minimize space used on disk
+* If a valid state exists, memory constraints must not prevent it from being loaded
+* Write overhead is small
+
+## Stratagy
+### Persist
+The broker first consolidates all payloads in to a hashmap, replacing the paload with a referance id in the state. This removes duplicate payloads that are created when 2 or more devices are subscribed to the same topic. This new structure is then converted to a binary stream. This stream is passed to the GZip algorythm and the resulting compressed binary is written directly to disk. This file is called a snapshot.
```

Comment:
I worth mention that once broker receive signal to persist state it makes a **whole copy** of its state,
consolidates payloads, and then persist. It doesn't not directly manipulate with broker state. It helps to avoid data races but may lead to confusion from operator point of view if not stated.